vms/platformvm/state/subnet_only_validator.go

```diff
@@ -22,9 +26,51 @@ var (
 	_ btree.LessFunc[SubnetOnlyValidator] = SubnetOnlyValidator.Less
 	_ utils.Sortable[SubnetOnlyValidator] = SubnetOnlyValidator{}
 
-	ErrMutatedSubnetOnlyValidator = errors.New("subnet only validator contains mutated constant fields")
+	ErrMutatedSubnetOnlyValidator     = errors.New("subnet-only validator contains mutated constant fields")
+	ErrConflictingSubnetOnlyValidator = errors.New("subnet-only validator contains conflicting subnetID + nodeID pair")
+	ErrDuplicateSubnetOnlyValidator   = errors.New("subnet-only validator contains duplicate subnetID + nodeID pair")
 )
 
+type SubnetOnlyValidators interface {
```

Comment:
Can we add compile time checks that `diff` and `state` implement `SubnetOnlyValidators`?

Reply:
1. There already exists a compile time check that `diff` implements `Diff`. `Diff` extends `Chain` which extends `SubnetOnlyValidators`.
2. There already exists a compile time check that `state` implements `State`. `Chain` extends `Chain` which extends `SubnetOnlyValidators`. 

So the checks _do_ exist... But could be more explicit. Do you want them added anyway?

Reply:
Nope, as long as it's there 🫡

Reply:
thanks for clarifying